File: mcspring-build/mcspring-plugin-manager/src/test/java/in/kyle/mcspring/manager/commands/TestCommandSpeed.java
package in.kyle.mcspring.manager.commands;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;

import in.kyle.mcspring.test.MCSpringTest;
import in.kyle.mcspring.test.command.TestCommandExecutor;
import in.kyle.mcspring.test.command.TestSender;

import static org.assertj.core.api.Assertions.*;
import static org.mockito.Mockito.*;

@MCSpringTest
class TestCommandSpeed {
    
    @Autowired
    TestCommandExecutor executor;
    
    TestSender sender;
    
    @BeforeEach
    void setup() {
        sender = spy(TestSender.class);
    }
    
    @Test
    void testSpeed() {
        executor.run(sender, "speed 10");
        assertThat(sender.getMessages()).first().asString().matches("Speed set to [^ ]+");
        verify(sender).setFlySpeed(10);
        verify(sender).setWalkSpeed(10);
    }
    
    @Test
    void testSpeedUsage() {
        executor.run(sender, "speed");
        assertThat(sender.getMessages()).first().asString().startsWith("Usage: ");
        verify(sender, never()).setFlySpeed(anyFloat());
        verify(sender, never()).setWalkSpeed(anyFloat());
    }
}

File: mcspring-build/mcspring-plugin-manager/src/test/java/in/kyle/mcspring/manager/commands/TestCommandSpeed.kt
package `in`.kyle.mcspring.manager.commands

import `in`.kyle.mcspring.test.MCSpringTest
import `in`.kyle.mcspring.test.command.TestCommandExecutor
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.mockito.ArgumentMatchers.anyFloat
import org.mockito.Mockito.*
import org.springframework.beans.factory.annotation.Autowired

@MCSpringTest
internal class TestCommandSpeed() {

    @Autowired
    lateinit var executor: TestCommandExecutor

    @Test
    fun testSpeed() {
        val (sender, messages) = executor.makeTestPlayer()
        executor.run(sender, "speed 10")
        assertThat(messages).first().asString().matches("Speed set to [^ ]+")
        verify(sender, times(1)).walkSpeed = 10F
        verify(sender, times(1)).flySpeed = 10F
    }

    @Test
    fun testSpeedUsage() {
        val (sender, messages) = executor.makeTestPlayer()
        executor.run(sender, "speed")
        assertThat(messages).first().asString().matches("Usage: .+")
        verify(sender, never()).walkSpeed = anyFloat()
        verify(sender, never()).flySpeed = anyFloat()
    }
}
